lib/cartopy/mpl/geoaxes.py

```diff
@@ -261,10 +261,36 @@ def set_boundary(self, path, transform):
         self.set_transform(transform)
         self.stale = True
 
+    @contextlib.contextmanager
+    def _cx_disable_path_siplification(self, path):
+        """Contextmanager to temporarily disable path simplification."""
+        try:
+            should_simplify = path.should_simplify
+            path.should_simplify = False
+            yield
+        finally:
+            path.should_simplify = should_simplify
```

Comment:
Now that I look at this, it seems like it would be simpler/shorter to just inline the try/finally block with the call below.